### PR TITLE
v0.4.2 was not yet finished

### PR DESCRIPTION
**Pull request by lightyear15**

### CODE REVIEW

**Review by jeff1010322 — CHANGES_REQUESTED** [review]
Could you also update the examples in the README to match the updates to the client?

Looks like it still references the removed readonlySubmit and has the old ABI format.

File: README.md [hunk 1]
```diff
@@ -106,7 +100,7 @@ async function demo() {
 
   /*
     {
-      accountInfo: { name: '', nonce: '144', permissionedKeys: [] },
+      accountInfo: { name: '', permissionedKeys: [] },
```
Comment (jeff1010322) [inline README.md:103]: The `name` field here should also be removed.

File: README.md [hunk 2]
```diff
@@ -282,22 +276,6 @@ async function demo() {
...
-
-  /*
-    {
-      result: 'Hello World!',
-      stateStatus: { previousBlock: '0', transactionCount: '161' },
-      status: 1,
-      statusInfo: 'Readonly request executed successfully.'
-    }
-  */
-}
```
Comment (jeff1010322) [inline README.md:300]: This closing bracket should stay.  It matches to the demo function block.

**Review by jeff1010322 — COMMENTED** [review]
There are a few other README changes that could be made:

- Timestamp should be removed from the block header comments.

- The example action can just set a static nonce value instead of using `res.nonce` which came from the previous nonce lookup example. something like: `nonce: "1"` would work.

- The TransactionSubmit commented output has changed.  Instead of:
```
{
      transactionID: '44b073a621b88df2345024b6584dee1421c44f05f6ddb50a94cbc5ff01c713ea',
      status: 1,
      statusInfo: 'Transaction has been accepted and is being executed.'
    }
```
it should be something like:
```
{
  transactionInfo: {
    enum: 2,
    value: '19812f6f13bd20da71a06d4be8e00b1b2a99d0cf5bf9254a484c51de544789de'
  },
  status: 1,
  statusInfo: 'Transaction has been accepted and is being executed.'
}
```

**Review by jeff1010322 — COMMENTED** [review]
Looks good, I think you just need to update the package.json version number to v0.4.3 now.

**Comment by lightyear15** [comment]
> Looks good, I think you just need to update the package.json version number to v0.4.3 now.

Hi @jeff1010322 ,
I have updated the version as well as the dependency on mazzaroth-xdr.
mazzaroth-js now depends on mazzaroth-xdr v0.4.3.
As we briefly discussed yesterday, in order to keep version numbers aligned, we should probably create a v0.4.3 of mazzaroth-xdr as well.

**Comment by jeff1010322** [comment]
> > Looks good, I think you just need to update the package.json version number to v0.4.3 now.
> 
> Hi @jeff1010322 ,
> I have updated the version as well as the dependency on mazzaroth-xdr.
> mazzaroth-js now depends on mazzaroth-xdr v0.4.3.
> As we briefly discussed yesterday, in order to keep version numbers aligned, we should probably create a v0.4.3 of mazzaroth-xdr as well.

I don't think we should go down the route of updating mazzaroth-xdr every time a library using it updates.  I think as long as the minor versions match `v0.4.*` we should be okay.  Otherwise, we are going to have to update all tools, Rust and Javascript that rely on this dependency even though nothing was changed.  I think the expectation should be that the latest minor versions should be compatible with our various tools, and they each can have their own bug fix/patches applied that give them different version numbers within that range.

File: src/client/contract-client.js [hunk 3]
```diff
@@ -117,7 +117,7 @@ class Client {
         const action = {
           channelID: this.channelID,
           nonce: nonce,
-          blockExpirationNumber: blockExpirationNumber,
+          blockExpirationNumber: blockExpirationNumber.toString(),
```
Comment (jeff1010322) [inline src/client/contract-client.js:120]: This makes sense, but broke a few of the tests:

/test/client/contract-client.js needs updated at lines 368, 390 and 413 to expect the string value for this field:
```
blockExpirationNumber: '11',
```

**Review by jeff1010322 — APPROVED** [review]

**Pull request merged**